builtin/fsmonitor--daemon.c

```diff
@@ -1311,9 +1311,15 @@ static int fsmonitor_run_daemon(void)
 	strbuf_addbuf(&state.path_gitdir_watch, &state.path_worktree_watch);
 	strbuf_addstr(&state.path_gitdir_watch, "/.git");
 	if (!is_directory(state.path_gitdir_watch.buf)) {
+		char *buf;
+		int len;
 		strbuf_reset(&state.path_gitdir_watch);
 		strbuf_addstr(&state.path_gitdir_watch,
 			      absolute_path(repo_get_git_dir(the_repository)));
+		buf = state.path_gitdir_watch.buf;
+		len = state.path_gitdir_watch.len;
+		if (len >= 2 && buf[len - 2] == '/' && buf[len - 1] == '.')
+			strbuf_setlen(&state.path_gitdir_watch, len - 2);
```

Comment:
How about using [`strbuf_strip_suffix()`](https://github.com/git/git/blob/v2.46.2/strbuf.h#L532-L539) instead?

```c
strbuf_strip_suffix(&state.path_gitdir_watch, "/.");
```

Also, did you want to add tests?

Reply:
Thank you! I'll update the commit.